crates/bevy_reflect/bevy_reflect_derive/src/derive_data.rs

```diff
@@ -409,9 +470,10 @@ impl<'a> ReflectStruct<'a> {
         let bevy_reflect_path = &self.meta().bevy_reflect_path;
         WhereClauseOptions {
             active_types: self.active_types().into(),
-            active_trait_bounds: quote! { #bevy_reflect_path::Reflect },
+            active_trait_bounds: quote! { #bevy_reflect_path::Reflect + #bevy_reflect_path::TypePath },
```

Comment:
this change is actually not necessary with the introduction of `parameter_{types,trait_bounds}`